### ISSUE TITLE
FR: add a button to force an "upload all"

### ISSUE THREAD

**Issue by londerella** [beradeep/whitehole#8]
First of all, fantastic idea. 
I set the auto periodic upload but didn't see many new files over a couple of days. It would be nice to have a button to force the upload of all/selected photos.
Is there a way to only select specific directories?
Thanks!

**Comment by beradeep** [comment]
Hi, don't worry, the auto periodic backup is set begin after an initial delay of atleast 1 day.

**Comment by londerella** [comment]
I'm not seeing any new upload and it has pass at least 4 days.

**Comment by beradeep** [comment]
Hi, are you sure about that? 🤔  Have you checked the cloud section? I hope you've ensured that auto backup is enabled in the settings and that your phone has been connected to the internet sufficiently?

**Comment by londerella** [comment]
Yes, checked both. Last file upload was July 17th. I'm always connected.

**Comment by Brian2po** [comment]
Same issue, inconsistent auto upload, the app is always running in the background(unrestricted battery optimization).

Hope to see in the next update to "select all" or "upload all". 😁

**Comment by beradeep** [comment]
> Yes, checked both. Last file upload was July 17th. I'm always connected.

hi, I would urge you to try out the latest release and check once again after altering the backup frequency. in the earlier releases, it was set to `weekly` by default. if it still doesn't work, I would like to know what's your device.

**Comment by londerella** [comment]
Already updated. I'll come back with more info

**Comment by beradeep** [comment]
> Same issue, inconsistent auto upload, the app is always running in the background(unrestricted battery optimization).
> 
> Hope to see in the next update to "select all" or "upload all". 😁

@Brian2po, the app won't always run in the background. It will only run when the backup task is running.

**Comment by ltkun** [comment]
same issue with  Xiaomi 13 ultra on a paranoid android.

**Comment by jojowakaki** [comment]
I recently had this, photos were not uploading. But it got fixed again, these are the things I tried, I don't know what exactly worked.
1. Turned off Auto periodic backup and turned it back on (Felt like this wasn't what made it work because now I when I look back, I think I did more steps because I felt like this didn't work)
2. Remade the connection with the bot

Followed the following step for this.
1. Exported database
2. Went to app info and cleared cache and storage.
3. Then it asks to remake the connection with bot
4. Go to bot father and get the bot code for you old bot
5. Paste that into the app
6. Go to your chat where you are sharing photos
7. type '/start' again
8. Copy that message (it is the same one but seems like that code needs to be the last message for whitehole to connect to the chat)
9. This should get your chat connected but you will be missing all the previous backed up images
10. Import database (this should populate all your cloud images again)
11. Re-enable backup (daily)
12. Wait a day, and walla

It would be nice to have something like a 'sync now' or 'backup now' button. It would be useful for people first installing an app with 2-3k pictures already on the phone, so you don't have to wait 24 hours to get your sync going.
